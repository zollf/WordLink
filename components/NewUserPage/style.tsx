import { StyleSheet } from "react-native";

import AppStyle from "../../styles";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
  },
  header: {
    flex: 1.5,
    justifyContent: "center",
    alignItems: "center",
  },
  input: {
    flex: 1,
    alignItems: "center",
  },
  button: {
    flex: 2.5,
    alignItems: "center",
  },
});

export default styles;
