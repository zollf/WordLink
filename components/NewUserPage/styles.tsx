import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
  },
  header: {
    flex: 1.5,
    justifyContent: "center",
    alignItems: "center",
  },
  input: {
    flex: 1,
    alignItems: "center",
  },
  button: {
    flex: 2.5,
    alignItems: "center",
  },
  username: {
    width: 229,
    fontSize: 12,
  },
  font: {
    fontSize: 24,
  },
});

export default styles;
