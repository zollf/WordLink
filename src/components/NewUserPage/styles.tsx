import { StyleSheet } from 'react-native';
import { heightPercentageToDP, widthPercentageToDP } from 'react-native-responsive-screen';
import AppStyle from 'app/src/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  header: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: heightPercentageToDP('-5%'),
    marginBottom: 25,
  },
  username: {
    width: 229,
    fontSize: 12,
  },
  text: {
    color: 'black',
    fontFamily: AppStyle.fontBold,
  },
  logotext: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    fontSize: heightPercentageToDP('4%'),
    textAlign: 'center',
    marginTop: heightPercentageToDP('5%'),
  },
  image: {
    height: 30,
    width: 30,
    transform: [{ rotate: '-90deg' }],
    marginBottom: 25,
  },
  bottom: {
    marginBottom: 50,
  },
  textWrapper: {
    width: widthPercentageToDP('75%'),
  },
});

export default styles;
