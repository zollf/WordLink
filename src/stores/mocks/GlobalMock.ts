const GlobalMock: Global = {
  _ready: true,
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
  setUserInfo: jest.fn(),
  userInfo: {
    username: 'jeff',
    points: 20,
    puzzles: {
      completed: 2,
      easy: [],
      medium: [],
      hard: [],
    },
    hintsUsed: 6,
    hardestDifficulty: 'Medium',
    starsEarnt: 69,
  },
  init: jest.fn(),
};

export default GlobalMock;
