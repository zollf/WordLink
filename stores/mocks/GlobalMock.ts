const GlobalMock: Global = {
  currentPage: 'index',
  setCurrentPage: jest.fn(),
  previousPage: 'index',
  currentDifficultyOpen: '',
  setCurrentDifficultyOpen: jest.fn((d: string) => (GlobalMock.currentDifficultyOpen = d)),
  getDifficultyColor: jest.fn(() => ''),
  setStartingPage: jest.fn(),
  onStartingPage: false,
};

export default GlobalMock;
