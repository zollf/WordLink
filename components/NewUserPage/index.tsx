import React, { useState } from "react";
import { Text, View, TextInput, TextBase } from "react-native";
import Button from "../Button";
import InputField from "../InputField";
import styles from "./style";

const NewUserPage = () => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.font}>Welcome Back!</Text>
        <Text style={styles.font}>Create a Username</Text>
      </View>
      <View style={styles.input}>
        <Text style={styles.username}>Username</Text>
        <InputField color="blue" />
      </View>
      <View style={styles.button}>
        <Button text="Continue ->" color="orange" onPress={() => ({})} />
      </View>
    </View>
  );
};

export default NewUserPage;
