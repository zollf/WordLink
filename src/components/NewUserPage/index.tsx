import React, { useState } from 'react';
import { Text, View, Image, NativeSyntheticEvent, TextInputChangeEventData, KeyboardAvoidingView } from 'react-native';
import { observer } from 'mobx-react';

import Button from '../Button';
import InputField from '../InputField';
import Logo from '../Logo';
import Arrow from '../../images/left-arrow.png';
import { useStore } from 'app/src/hooks';
import _saveData from 'app/src/lib/saveData';

import styles from './styles';

const NewUserPage = () => {
  const [inputVal, setInputVal] = useState('');
  const { setUserInfo, setStartingPage } = useStore('global');

  const onChange = (e: NativeSyntheticEvent<TextInputChangeEventData>) => {
    setInputVal(e.nativeEvent?.text);
  };

  const onSubmit = async () => {
    if (!inputVal) {
      return;
    }
    const newUser: UserData = {
      username: inputVal,
      level: 1,
      points: 0,
      puzzles: {
        completed: 0,
        easy: [],
        medium: [],
        hard: [],
      },
      hintsUsed: 0,
      hardestDifficulty: 'None',
      starsEarnt: 0,
    };

    await _saveData(newUser)
      .then((v) => {
        if (v.success) {
          setStartingPage(false);
          setUserInfo(newUser);
        }
      })
      .catch();
  };

  return (
    <KeyboardAvoidingView style={styles.container} behavior="padding">
      <View style={styles.header}>
        <Logo />
        <View style={styles.textWrapper}>
          <Text style={[styles.text, styles.logotext]}>Enter a Username!</Text>
        </View>
      </View>
      <Image source={Arrow} style={styles.image} />
      <View style={styles.bottom}>
        <Text>Username</Text>
        <InputField color="blue" value={inputVal} onChange={onChange} />
      </View>
      <View>
        <Button text="Continue" color="orange" onPress={onSubmit} />
      </View>
    </KeyboardAvoidingView>
  );
};

export default observer(NewUserPage);
