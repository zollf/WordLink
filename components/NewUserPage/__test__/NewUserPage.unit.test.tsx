import React from 'react';
import { Provider } from 'mobx-react';
import GlobalMock from '../../../stores/mocks/GlobalMock';

import NewUserPage from '../';

import renderer from 'react-test-renderer';

const component = (
  <Provider global={GlobalMock}>
    <NewUserPage />
  </Provider>
);

describe('<NewUserPage />', () => {
  const wrapper = renderer.create(component);
  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('clicks button works correctly', () => {
    wrapper.root.findByProps({ 'data-test-id': 'button' }).props.onPress();
    //todo test
  });
});
