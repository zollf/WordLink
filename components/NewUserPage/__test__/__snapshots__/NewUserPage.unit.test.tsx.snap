// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewUserPage /> matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1.5,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Welcome Back!
    </Text>
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Create a Username
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 12,
          "width": 229,
        }
      }
    >
      Username
    </Text>
    <TextInput
      allowFontScaling={true}
      data-test-id="text-input"
      editable={true}
      onChange={[Function]}
      rejectResponderTermination={true}
      selectTextOnFocus={true}
      style={
        Array [
          Object {
            "borderRadius": 5,
            "height": 93.5,
            "shadowColor": "#000",
            "shadowRadius": 4,
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "#40BAFF",
            "width": 450,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2.5,
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": mockConstructor {},
            },
          ],
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                4294956160,
                4294950464,
              ],
              "endPoint": undefined,
              "locations": Array [
                0.5,
                0.5,
              ],
              "startPoint": Array [
                -0.1,
                0.1,
              ],
            }
          }
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "display": "flex",
              "height": 106.5,
              "justifyContent": "center",
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 4,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "fontFamily": "Raleway_400Regular",
                "fontSize": 26.5,
              }
            }
          >
            Continue
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </View>
</View>
`;
